core/src/main/java/org/eclipse/hono/config/AbstractConfig.java

```diff
@@ -91,6 +103,45 @@ public AbstractConfig(final AbstractConfig other) {
         this.trustStorePath = other.trustStorePath;
     }
 
+    /**
+     * Gets the password represented by a property value.
+     * <p>
+     * This method determines the password as follows:
+     * <ol>
+     * <li>If the given value does not start with {@value #PREFIX_FILE}, then the password is the given value.</li>
+     * <li>Otherwise the password is the UTF-8 encoded string represented by the first line read from the
+     * file indicated by the property's value after the {@value #PREFIX_FILE} prefix.</li>
+     * </ol>
+     *
+     * @param purpose A (very) short description of the context in which the password is being used.
+     * @param value The property value to determine the password from.
+     * @return The password.
+     */
+    protected final String getPassword(final String purpose, final String value) {
+
+        Objects.requireNonNull(purpose);
+        Objects.requireNonNull(value);
+
+        if (!value.startsWith(PREFIX_FILE)) {
+            return value;
+        }
+
+        final String fsPath = value.substring(PREFIX_FILE.length());
+        final File file = new File(fsPath);
+        if (!file.exists()) {
+            LOG.warn("cannot read {} password, file [{}] does not exist", purpose, fsPath);
```

Comment:
What if an (existing) password starts by accident with "file:"? (I admit that this is not very likely). One could return the value as a literal password if the file is not found. The downside would be of course that you don't have such a clear result in error cases.

Reply:
Then it doesn't work :-) I am willing to take that risk. BTW the behavior is documented so users should be able to prevent this from happening by simply choosing a more appropriate password for their key store ...